### PR TITLE
bugfix/#103-gameengine-increased

### LINKED ISSUE #103: Make the GameEngine more testing-friendly
Allow initializing GameEngine with
* [x] custom player initial position
* [x] custom width and height

### PR DESCRIPTION
**Pull request by DorianRec**
GameEngine now allows a custom initial player position,
custom playing area sizes and
GameEngine were cleared up a bit.

* [x] Hider and Searcher need to know the playing area dimension.
* [x] Hider may give treasure positions, which lay outside the playing area.

Fixes #103

### CODE REVIEW

File: src/main/java/com/treasure/hunt/game/GameEngine.java [hunk 1]
```diff
@@ -30,37 +28,81 @@
...
     protected final Searcher searcher;
     protected final Hider hider;
+    protected final Coordinate initialSearcherCoordinate;
+
     /**
      * Tells, whether the game is done or not.
      */
     @Getter
-    private BooleanProperty finished = new SimpleBooleanProperty(false);
+    protected boolean finished = false;
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/game/GameEngine.java:47]: sorry but this is needed to be a Property so we can bind ui on it

File: src/main/java/com/treasure/hunt/game/GameEngine.java [hunk 2]
```diff
@@ -30,37 +28,81 @@
...
      */
-    public static final int HEIGHT = 200;
+    @Getter
+    protected final int height;
     /**
      * The width of the playing area.
      */
-    public static final int WIDTH = 200;
+    @Getter
+    protected final int width;
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/game/GameEngine.java:37]: We need to tell the hider and searcher but this value

File: src/main/java/com/treasure/hunt/game/GameManager.java [hunk 3]
```diff
@@ -48,7 +48,7 @@
     private ObservableList<Move> moves = FXCollections.observableArrayList();
 
     private GameEngine gameEngine;
-
+    private BooleanProperty finished = new SimpleBooleanProperty(false);
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/game/GameManager.java:53]: ok if u do this u can make the other one a boolean

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
I did not test it make yure u do before merging

File: src/main/java/com/treasure/hunt/game/GameEngine.java (hunk 2 above)
Comment (DorianRec) [reply src/main/java/com/treasure/hunt/game/GameEngine.java:37]: This is right, I'm going to fix this

**Review by 1-alex98 — DISMISSED** [review]

**Review by 1-alex98 — DISMISSED** [review]

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**